lib/active_resource/threadsafe_attributes.rb

```diff
@@ -0,0 +1,57 @@
+module ThreadsafeAttributes
+  def self.included(klass)
+    klass.extend(ClassMethods)
+  end
+
+  module ClassMethods
+    def threadsafe_attribute(*attrs)
+      attrs.each do |attr|
+        define_method attr do
+          get_threadsafe_attribute(attr)
+        end
+
+        define_method "#{attr}=" do |value|
+          set_threadsafe_attribute(attr, value)
+        end
+
+        define_method "#{attr}_defined?" do
+          threadsafe_attribute_defined?(attr)
+        end
+      end
+    end
+  end
+
+  private
+
+  def get_threadsafe_attribute(name)
+    if threadsafe_attribute_defined_by_thread?(name, Thread.current)
+      get_threadsafe_attribute_by_thread(name, Thread.current)
+    elsif threadsafe_attribute_defined_by_thread?(name, Thread.main)
+      get_threadsafe_attribute_by_thread(name, Thread.main).dup.tap do |value|
```

Comment:
The five values it's being used for are: headers, connection, user, password, site, and proxy

user and proxy are `String`
site is a `URI`
headers is a `Hash` with plain string keys/values
connection seems to be a simple object that references other simple objects

I guess the best argument is that we've been using these changes in production for a while with no threadsafety issues, but I suppose we tend to be using them in the same way every time.